cirq/ops/thermal_relaxation_channel.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for the thermal relaxation channel."""
+
+from typing import Iterable
+
+import numpy as np
+
+from cirq import protocols, value
+from cirq.ops import gate_features
+
+
+@value.value_equality
+class ThermalRelaxationChannel(gate_features.SingleQubitGate):
+    """Dampen qubits at varying rates through non ideal disipation.
+
+    This channel models the effect of energy dissipation into the environment
+    as well as the environment depositing energy into the system at varying
+    rates.
+    """
+
+    def __init__(self, p: float, gamma: float, beta: float) -> None:
```

Comment:
Rename this to describe what they actually are, instead of referring to the mathematical symbols. E.g. rename to `gamma` to `dissipate_probability`.

Reply:
all other channels follow the construction I have here and in the docstring I show some matrices involving `GeneralizedAmplitudeDampingChannel` and having the variables have the same name makes things pretty clear. Are you sure you want me to change them ?

Reply:
We shouldn't have done that there. It's more serious here because the parameters are more conceptually similar. Rename 'em.

Reply:
Will do.